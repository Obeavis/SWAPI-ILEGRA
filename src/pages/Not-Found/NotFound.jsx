import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import stormtrooper from "../../assets/images/stormtrooper.png";
import "./NotFound.scss";

const NotFound = () => (
	<Container className="not-found">
		<Row>
			<Col lg={6} className="text-center text-lg-right">
				<h2 className="text-20 text-bold">404</h2>

				<p className="text-white text-12  text-uppercase">
					This is not the page you're looking for
				</p>
			</Col>
			<Col lg={6} className="text-center text-lg-left">
				<img
					alt="stormtrooper"
					src={stormtrooper}
					className="img-fluid"
				/>
			</Col>
		</Row>
	</Container>
)

export default NotFound;