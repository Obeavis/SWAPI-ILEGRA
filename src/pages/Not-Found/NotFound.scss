.not-found {
  margin-top: 150px;

  h2 {
    color: #f2bc00;
  }

  img {
    margin-left: auto;
    margin-right: auto;
    margin-top: 3.5rem;
  }
}
