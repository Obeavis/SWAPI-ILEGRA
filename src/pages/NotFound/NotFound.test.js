import React from "react";
import { MemoryRouter } from 'react-router-dom';
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import NotFound from "./NotFound";

configure({ adapter: new Adapter() });

describe("Film Component", () => {

	const component = (
		<MemoryRouter>
			<NotFound />
		</MemoryRouter>
	)

	it("matches the snapshot", () => {
		const tree = renderer.create(component).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("Renders the right image", () => {
		const wrapper = mount(component);
		const image = wrapper.find('img').prop("src");

		expect(image).toEqual('stormtrooper.png');

	});

	it("Have the right text inside", () => {
		const wrapper = mount(component);
		const error = wrapper.find('h2').text();
		const text = wrapper.find('p').text();

		expect(error).toEqual('404');
		expect(text).toEqual("This is not the page you're looking for");
	});

})