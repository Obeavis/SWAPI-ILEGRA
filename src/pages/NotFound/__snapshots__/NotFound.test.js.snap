// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Component matches the snapshot 1`] = `
<div
  className="not-found container"
>
  <div
    className="row"
  >
    <div
      className="text-center text-lg-right col-lg-6"
    >
      <h2
        className="text-20 text-bold"
      >
        404
      </h2>
      <p
        className="text-white text-12  text-uppercase"
      >
        This is not the page you're looking for
      </p>
    </div>
    <div
      className="text-center text-lg-left col-lg-6"
    >
      <img
        alt="stormtrooper"
        className="img-fluid"
        src="stormtrooper.png"
      />
    </div>
  </div>
</div>
`;
